### PR TITLE
client-go/cache: fix the AddIfNotPresent method of the DeltaFIFO

### PR DESCRIPTION
**Pull request by Iceber**
Signed-off-by: Iceber Gu <wei.cai-nat@daocloud.io>

#### What type of PR is this?

/kind bug
#### What this PR does / why we need it:
if the obj is `Deltas{}`, `deltas.Newest()` will return nil, and then panic

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
None
```

### CODE REVIEW

**Comment by caesarxuchao** [comment]
Can you add a unit test?

/assign @yliaog 
/triage accepted

**Comment by Iceber** [comment]
fix nested KeyError: https://github.com/kubernetes/kubernetes/pull/100768

**Comment by Iceber** [comment]
@caesarxuchao  added a unit test, but the nested KeyError needs to be fixed first.
https://github.com/kubernetes/kubernetes/pull/100768

File: staging/src/k8s.io/client-go/tools/cache/delta_fifo_test.go [hunk 1]
```diff
@@ -541,6 +541,12 @@ func TestDeltaFIFO_detectLineJumpers(t *testing.T) {
 func TestDeltaFIFO_addIfNotPresent(t *testing.T) {
 	f := NewDeltaFIFO(testFifoObjectKeyFunc, nil)
 
+	emptyDeltas := Deltas{}
+	expectedErr := KeyError{interface{}(emptyDeltas), ErrZeroLengthDeltasObject}
```
Comment (yliaog) [inline staging/src/k8s.io/client-go/tools/cache/delta_fifo_test.go:545]: nested error is expected in this case.

Comment (Iceber) [reply staging/src/k8s.io/client-go/tools/cache/delta_fifo_test.go]: @yliaog updated

**Comment by Iceber** [comment]
updated to use the `errors.Is` in the test to determine `err`
@yliaog  PTAL, Thanks

**Pull request merged**